cvat/apps/quality_control/quality_reports.py

```diff
@@ -1000,6 +1004,20 @@ def _match_ann_type(self, t, *args):
         else:
             return None
 
+    def match_labels(self, item_a, item_b):
+        def label_distance(a, b):
+            if a is None or b is None:
+                return 0
+            return 0.5 + (a.label == b.label) / 2
```

Comment:
Is this formula better than simple `a.label == b.label`?

Reply:
- Case 1: **One of the labels isn't present**, then in such case, the distance should be `0`
- Case 2: When there's an **intersection between a set of labels**, they should have a distance of `1`
- Case 3: There's a **mismatch of labels**, i.e., a case between the above cases. The distance can be anything be 0 to 1 but fixed across all labels here `0.5`

Wherever there's `distance` in mind try to make an analogy with IoU in case of shape.

https://github.com/cvat-ai/cvat/pull/7582#discussion_r1550197708 based on our conversation in this, I got this idea.

Reply:
Ok, the question is only about the last line, so let's discuss the point 3. I don't think we can really use any constant, because if it's less than `self.iou_thresold`, it can be filtered out during matching. Probably, it must be bigger.

Reply:
If that constant value is equal to or more than `self.iou_threshold`, then we are good to go. One thing we can do is to pass another named parameter in the `self._match_segments` function related to `dist_thresh` by this, we can fix the dist_thresh constant `0.5` for the label case. 

How come you are able to think so so far-fetched relations? I wouldn't have realised this until somebody raised an issue that after raising the iou_threshold the results are coming incorrect.

Reply:
Hello @zhiltsov-max a gentle reminder to this.

Reply:
Yes, probably we can fix the threshold to some value for this check, as it's not really applicable to tag comparison. Maybe it will be better to return back to the non-class `_match_segments` then, or add a function parameter.